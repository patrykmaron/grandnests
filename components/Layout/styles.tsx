import styled from 'styled-components'

export const Wrapper = styled.div`
  display: grid;
  grid-template-rows: auto 1fr auto;
  min-height: 100vh;
`
