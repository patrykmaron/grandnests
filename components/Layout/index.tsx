import * as React from 'react'
import Head from 'next/head'
import { Wrapper } from './styles'

type Props = {
  title?: string
}

const Layout: React.FC<Props> = ({
  children,
  title = 'Grand Nests | Default Title',
}) => (
  <>
    <Head>
      <title>{title}</title>
      <meta charSet="utf-8" />
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
    </Head>
    <Wrapper>{children}</Wrapper>
  </>
)

export default Layout
