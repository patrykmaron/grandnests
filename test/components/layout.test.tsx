import React from 'react'
import { render } from '../testUtils'
import Layout from '../../components/Layout'
import { cleanup, waitFor } from '@testing-library/react'

afterEach(cleanup)

describe('Layout component', () => {
  it('matches snapshot and renders', () => {
    const { asFragment } = render(<Layout />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
