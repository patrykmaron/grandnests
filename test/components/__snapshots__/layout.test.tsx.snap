// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component matches snapshot and renders 1`] = `
<DocumentFragment>
  <div
    class="styles__Wrapper-kt6arw-0 lcCCSV"
  />
</DocumentFragment>
`;
